libd/src/d/semantic/symbol.d

```diff
@@ -738,7 +741,33 @@ struct SymbolAnalyzer {
 		
 		c.step = Step.Processed;
 	}
-	
+
+	void analyze (InterfaceDeclaration d, Interface i){
+
+		auto oldManglePrefix = manglePrefix;
+		auto oldScope = currentScope;
+		auto oldThisType = thisType;
+		auto oldMethodIndex = methodIndex;
+		
+		scope(exit) {
+			manglePrefix = oldManglePrefix;
+			currentScope = oldScope;
+			thisType = oldThisType;
+			methodIndex = oldMethodIndex;
+		}
+
+		thisType = Type.get(i).getParamType(false, true);
+
+		import std.conv;
+		auto name = i.name.toString(context);
+		manglePrefix = manglePrefix ~ to!string(name.length);
+
```

Comment:
Make sure you fail on what is not supported, not go through ignoring the problem. In that case, make sure the interface has no member and does inherit from anything. If the interface does, it should be an assertion failure.

Not supporting everything is fine, but ignoring it is not an option.